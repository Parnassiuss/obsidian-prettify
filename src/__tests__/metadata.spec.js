const prettifier = require('../prettifier')

test("Creates a header", () => {
    const content = `No Header`
    return prettifier(content, {
        alwaysCreateHeader: true,
        lastModifiedAt: 'Sun, 21 Nov 2020 21:34:48 GMT'
    }).then(data => {
        expect(data).toMatchSnapshot();
    });
});


test("Maintains properties in header", () => {
    const content = `---
title: 'Example'
list:
  - oneX
  - 0
  - false
---
# What the Semantic Web ignores
    
The Web is not only about facts,
It's about humans communicating all sorts of things.
In their own language, with their own model of the world.

perhaps 1% ? of the humans use it for facts?
            `
    return prettifier(content, {
        lastModifiedAt: 'Sun, 21 Nov 2020 21:34:48 GMT'
    }).then(data => {
        expect(data).toMatchSnapshot();
    });
});


test("Respects hash", () => {
    const content = `---
tag: '#Interpretability'
---
            `
    return prettifier(content, {
        lastModifiedAt: 'Sun, 21 Nov 2020 21:34:48 GMT'
    }).then(data => {
        expect(data).toMatchSnapshot();
    });
});




