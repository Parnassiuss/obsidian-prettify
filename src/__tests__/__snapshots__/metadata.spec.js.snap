// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a header 1`] = `
VFile {
  "contents": "---
lastModifiedAt: 'Sun, 21 Nov 2020 21:34:48 GMT'
---

No Header
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Maintains properties in header 1`] = `
VFile {
  "contents": "---
title: Example
list:
  - oneX
  - 0
  - false
lastModifiedAt: 'Sun, 21 Nov 2020 21:34:48 GMT'
---

# What the Semantic Web ignores

The Web is not only about facts,
It's about humans communicating all sorts of things.
In their own language, with their own model of the world.

perhaps 1% ? of the humans use it for facts?
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Respects hash 1`] = `
VFile {
  "contents": "---
tag: '#Interpretability'
lastModifiedAt: 'Sun, 21 Nov 2020 21:34:48 GMT'
---
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;
